### DISCUSSION TITLE
Extend Command line tab-completion to scripts inside nearest package.json

### DISCUSSION THREAD

**Discussion by madrussa** [pnpm/pnpm#4766] [Ideas]
Hello,

The https://pnpm.io/completion is a really nice touch. It would be reallllllly nice however if this could extend to tab-completing the nearest package.json scripts. So doing `pnpm \t` could in addition output `start` `test` `lint`... or if it may become too verbose having them under the run command would be ideal `pnpm run\t` would then give you the scripts list or options with scripts prioritised.

**Comment by madrussa** [comment]
apparently it is meant to do just that, but doesn't work for me. Could be a compatibility issue with oh-my-zsh

**Reply by madrussa** [reply]
Clear comp cache... works. Facepalm.

**Reply by CobyPear** [reply]
Hi @madrussa, I am also not seeing my local `package.json` scripts when trying to use tab complete. Can you explain which cache you had to clear? Thanks!

**Reply by madrussa** [reply]
Hi, I use oh-my-zsh so it may differ for you. But I followed the instructions listed here https://github.com/ohmyzsh/ohmyzsh/wiki/FAQ#how-do-i-reset-the-completion-cache

**Reply by CobyPear** [reply]
Thanks! It started working for me eventually :)

**Comment by michakfromparis** [comment]
Hey @madrussa @CobyPear @indexzero @weyert @cpojer,

I was looking for better Zsh completions for pnpm myself, especially for package.json scripts - the way I had it setup for yarn - but couldn’t find anything satisfying. So, I went ahead and built a plugin to handle it. Check it out here: [michakfromparis/zsh-pnpm-completions](https://github.com/michakfromparis/zsh-pnpm-completions).

I threw a bunch of aliases and live npm package completion on top while at it.

Hope this helps you guys too! Let me know if this works for you.

Cheers,

@michakfromparis
